inst/include/RcppArmadilloAs.h

```diff
@@ -403,14 +403,68 @@ namespace traits {
                     res = symmatl(res);
                 }
             }
+            else if (type == "pMatrix") {
+                IntegerVector i;
+                IntegerVector p(ncol + 1);
+                IntegerVector x(ncol, 1);
+                IntegerVector perm = mat.slot("perm");
+                
+                // Sort the row number by the column number
+                std::map <int, int> colrow;
+                for(int tmp = 0; tmp < perm.size(); tmp++){
+                    colrow[perm[tmp]] = tmp;
+                }
+                
+                // Calculate i
+                for(std::map<int, int>::iterator tmp = colrow.begin(); tmp != colrow.end(); tmp++){
+                    i.push_back(tmp -> second);
```

Comment:
Using `.push_back` is problematic on `IntegerVector` because Rcpp are proxy models that serve as shallow wrappers around R objects. Thus, each time you add an element, there is a significant overhead associated with copying the existing vector to new vector. It would be much better in this case to directory use `std::vector<int>` and then cast at the end to `IntegerVector`.